vertica_python/vertica/connection.py

```diff
@@ -566,19 +578,32 @@ def establish_socket_connection(self, address_list):
 
         # Failover: loop to try all addresses
         while addrinfo:
-            (family, socktype, proto, canonname, sockaddr) = addrinfo
+            if not self.proxy:
+                (family, socktype, proto, canonname, sockaddr) = addrinfo
+                # _AddressList filters all addrs to AF_INET and AF_INET6, which both
+                # have host and port as values 0, 1 in the sockaddr tuple.
+                host = sockaddr[0]
+                port = sockaddr[1]
+            else:
+                (family, socktype, proto, canonname, sockaddr, host, port) = addrinfo
             last_exception = None
 
-            # _AddressList filters all addrs to AF_INET and AF_INET6, which both
-            # have host and port as values 0, 1 in the sockaddr tuple.
-            host = sockaddr[0]
-            port = sockaddr[1]
-
             self._logger.info('Establishing connection to host "{0}" on port {1}'.format(host, port))
 
             try:
                 raw_socket = self.create_socket(family)
-                raw_socket.connect(sockaddr)
+                if self.proxy:
+                    self._logger.info(f'Connecting to proxy: {self.proxy}')
+                    proxy_tuple=(self.proxy.rsplit(':', 1)[0], int(self.proxy.rsplit(':', 1)[1]))
+                    raw_socket.connect(proxy_tuple)
+                    fp = raw_socket.makefile(mode='rw')
+                    fp.write('CONNECT %s:%d HTTP/1.0\r\n\r\n' % (host, port))
```

Comment:
I have updated with HTTP/1.1 and also added response status check. Please check
What do you think shall we add auth in this PR or do a new PR?
Never saw SSL for a proxy (since that will be local). I could be wrong
I don't have a definitive answer for this. For now I think "NO". I don't have a way to test against SOCKS proxy as of now. Perhaps, we shall update the README and clear that works with CONNECT method for now